README
NE LISEZ PAS CE README !!!!!!!!!!!!!!!!!!!!
il est beaucoup trop vieux, voir plutot http://hules.free.fr/wmcoincoin


wmCoinCoin 2.1beta: yet another braindamaged dock applet

-------------------<abstract>---------------------------------
 prerequis 
	- rien de particulier (x11,...)
 compilation: 
	./configure && make ; su ; make install	
 avant de lancer:
	wmcoincoin -h
--------------------</abstract>--------------------------------

Tout d'abord, ce programme n'a d'interet(!?!) que pour les personnes qui
cliquent 'refresh' 500 fois par jour sur DaLinuxFrenchPage (http://linuxfr.org)

Il est maintenant capable de fonctionner avec d'autres site que linuxfr.org,
sous reserve qu'ils tournent sous dacode 1.2 ou mieux.

c'est une dock applet pour windowmaker (http://www.wimdowmaker.org). 
Ses fonctionnalités::
 * Affichage de l'heure du dernier post sur la tribune
 * Affichage défilant des titres des dernières news de http://linuxfr.org
 * Visionnage rapide du contenu d'une news (sans les commentaires)
 * Troll-o-metre incorporé de qualité professionnelle.
 * Troll-o-scope de dernière génération, modele couleur multiprecision 55x21
 * (NEW!) Palmipède Editor, permet d'editer les messages a envoyer sur la tribune
ainsi que le useragent.
 * (NEW!) Estimation du nombre de personnes présentes sur la tribune.
 * (NEW!) Compatibilité KDE .
(faire: Menu KDE/Configurer/Ajouter/Extension/Applets Windowmaker),
 * le pinnipede etc...
 * non, le coredump ne fait pas partie des fonctionnalites, alors si
wmcoincoin a tendance a en pondre, faites le moi savoir :)

Pour plus de details, consulter la wmCoinCoinHomePage 
(http://hules.free.fr/wmcoincoin)

Pour la compilation: les informations classiques sont dans le fichier INSTALL.
* la libwraster n'est plus necessaire depuis la v2.2beta
* il n'y a PLUS BESOIN de curl, wmcoincoin s'est doté d'un magnifique module http, que 
  j'espere a peu pres fonctionnel..
* un DISPLAY sur au moins 15 bits (eh oui...)
* un BRAS, muni d'une main prolongée par un périphérique de pointage.

En option: 
* un cerveau (vraiment pas obligatoire :-)

pour compiler:

[prière à St Isodore, notre Saint Patron]

./configure

[à ce stade, je recommande de faire une offrande a shivah, ça peut aider]

make
make install

les options utiles de ./configure :
--prefix chemin (par defaut chemin == /usr/local )

Pour voir les options de wmCoinCoin, executer 'wmcoincoin -h'

[Nouveau (v2.x)]
* un fichier permettant de stocker les options (utile pour KDE qui ne sauve pas les options de
ligne de commande d'une session à l'autre) est utilisé: $HOME/wmcoincoin/options
(un fichier par défaut est crée au premier lancement, vous pouvez ensuite l'éditer)
Il contient aussi des options supplémentaires permettant de paramètrer le PINNIPEDE TELETYPE.


NEWS
2001-12-16 v2.3.2: bugfixes divers 
2001-12-02 super ispell pour le palmipede + bugfixes + cleanup pour le cvs
2001-10-28 la v2.2beta a aussi une bone tete
2001-09-29 la v2.1beta a une bonne tete
2001-09-19 rolala ça fait longtemps que j'ai pas touché a ce fichier.
2001-05-24 apres la 1.0beta, solide comme un roc, on va passer directement a la 1.1...
           Avec tout plein de nouveautés pour encore mieux troller:
		* des petits ballons d'aide, c'est pas utile mais c'est marrant a faire.
		* des messages dans de vrai fenetre quand il y a une erreur embetante
		* j'ai viré curl ! c'etait surtout histoire d'ecrire un petit truc 
		pour faire des GET et des POST, avec des sockets et tout.. instructif. 
		j'ai peut etre rendu le truc moins robuste qu'avec curl, mais c'est plus
		propre, et ca permet de mieux gerer les situations d'erreurs.
		* le palmipede editor, qui permet d'editer les messages/useragent a poster
		avec gestion des balises html...
		* la compatibilité KDE ! et sans doute plein d'autres windowmanagers, mais
		j'ai pas essayé.
2001-03-10 Ca fait des semaines que je me dis qu'il faut sortir la version suivante
	   Je crois que je vqis pqsser a l'acte.
2001-02-21 Je viens de corriger 2-3 conneries dans le README
2001-02-21 Apres reflexion, c'etait debile d'utiliser des threads --> retour aux
           bon vieux signaux. ca marche bien :-)
2001-02-19 La version 0.9 de wmCoinCoin semble prête !


src/md5.h
/*
  Copyright (C) 1999, 2002 Aladdin Enterprises.  All rights reserved.

  This software is provided 'as-is', without any express or implied
  warranty.  In no event will the authors be held liable for any damages
  arising from the use of this software.

  Permission is granted to anyone to use this software for any purpose,
  including commercial applications, and to alter it and redistribute it
  freely, subject to the following restrictions:

  1. The origin of this software must not be misrepresented; you must not
     claim that you wrote the original software. If you use this software
     in a product, an acknowledgment in the product documentation would be
     appreciated but is not required.
  2. Altered source versions must be plainly marked as such, and must not be
     misrepresented as being the original software.
  3. This notice may not be removed or altered from any source distribution.

  L. Peter Deutsch
  ghost@aladdin.com

 */
/* $Id: md5.h,v 1.1 2004/02/29 15:01:19 pouaite Exp $ */
/*
  Independent implementation of MD5 (RFC 1321).

  This code implements the MD5 Algorithm defined in RFC 1321, whose
  text is available at
	http://www.ietf.org/rfc/rfc1321.txt
  The code is derived from the text of the RFC, including the test suite
  (section A.5) but excluding the rest of Appendix A.  It does not include
  any code or documentation that is identified in the RFC as being
  copyrighted.

  The original and principal author of md5.h is L. Peter Deutsch
  <ghost@aladdin.com>.  Other authors are noted in the change history
  that follows (in reverse chronological order):

  2002-04-13 lpd Removed support for non-ANSI compilers; removed
	references to Ghostscript; clarified derivation from RFC 1321;
	now handles byte order either statically or dynamically.
  1999-11-04 lpd Edited comments slightly for automatic TOC extraction.
  1999-10-18 lpd Fixed typo in header comment (ansi2knr rather than md5);
	added conditionalization for C++ compilation from Martin
	Purschke <purschke@bnl.gov>.
  1999-05-03 lpd Original version.
 */

#ifndef md5_INCLUDED
#  define md5_INCLUDED

/*
 * This package supports both compile-time and run-time determination of CPU
 * byte order.  If ARCH_IS_BIG_ENDIAN is defined as 0, the code will be
 * compiled to run only on little-endian CPUs; if ARCH_IS_BIG_ENDIAN is
 * defined as non-zero, the code will be compiled to run only on big-endian
 * CPUs; if ARCH_IS_BIG_ENDIAN is not defined, the code will be compiled to
 * run on either big- or little-endian CPUs, but will run slightly less
 * efficiently on either one than if ARCH_IS_BIG_ENDIAN is defined.
 */

typedef unsigned char md5_byte_t; /* 8-bit byte */
typedef unsigned int md5_word_t; /* 32-bit word */

/* Define the state of the MD5 Algorithm. */
typedef struct md5_state_s {
    md5_word_t count[2];	/* message length in bits, lsw first */
    md5_word_t abcd[4];		/* digest buffer */
    md5_byte_t buf[64];		/* accumulate block */
} md5_state_t;

#ifdef __cplusplus
extern "C" 
{
#endif

/* Initialize the algorithm. */
void md5_init(md5_state_t *pms);

/* Append a string to the message. */
void md5_append(md5_state_t *pms, const md5_byte_t *data, int nbytes);

/* Finish the message and return the digest. */
void md5_finish(md5_state_t *pms, md5_byte_t digest[16]);

#ifdef __cplusplus
}  /* end extern "C" */
#endif

#endif /* md5_INCLUDED */
